File: algorithmic_toolbox/week2_algorithmic_warmup/4_least_common_multiple/lcm.go
package main

import (
	"C"
	"fmt"
)

func gcd(a, b int) int {
	var tmp int
	for b != 0 {
		tmp = b
		b = a % b
		a = tmp
	}
	return a
}

func lcm(a, b int) int64 {
	return int64(a) * int64(b) / int64(gcd(a, b))
}

func main() {
	var a, b int
	fmt.Scan(&a, &b)
	fmt.Println(lcm(a, b))
}

File: algorithmic_toolbox/week2_algorithmic_warmup/4_least_common_multiple/lcm.c
#include <stdio.h>

int gcd(int a, int b) {
  int tmp;
  while (b != 0) {
    tmp = b;
    b = a % b;
    a = tmp;
  }
  return a;
}

long long lcm(int a, int b) {
  return ((long long)a * b) / gcd(a, b);
}

int main() {
  int a, b;
  int _ = scanf("%d%d", &a, &b);
  printf("%lld\n", lcm(a, b));
  return 0;
}
